packages/modular-scripts/src/utils/config.ts

```diff
@@ -0,0 +1,98 @@
+import { cosmiconfigSync } from 'cosmiconfig';
+import path from 'path';
+import getModularRoot from './getModularRoot';
+
+// Where cosmiconfig can look for the configuration
+const searchPlaces = [
+  '.modular.js',
+  'package.json',
+  `.modularrc`,
+  `.modularrc.json`,
+  `.modularrc.yaml`,
+  `.modularrc.yml`,
+  `.modularrc.js`,
+  `.modularrc.cjs`,
+  `modular.config.js`,
+  `modular.config.cjs`,
+];
+
+/**
+ * Configuration file interface
+ */
+interface Config {
+  useModularEsbuild: boolean;
+  externalCdnTemplate: string;
+  externalBlockList: string[];
+  externalAllowList: string[];
+  publicUrl: string;
+  generateSourceMap: boolean;
+}
+
+type ConfigDefs = {
+  [Key in keyof Config]: {
+    default: Config[Key];
+    override: undefined | Config[Key];
+  };
+};
+
+/**
+ * Defaults and env variable overrides
+ */
+const defs: ConfigDefs = {
+  useModularEsbuild: {
+    default: false,
+    override:
+      process.env.USE_MODULAR_ESBUILD === 'true' ||
+      process.env.USE_MODULAR_ESBUILD === 'false'
+        ? process.env.USE_MODULAR_ESBUILD === 'true'
+        : undefined,
+  },
+  externalCdnTemplate: {
+    default: 'https://esm.sh/[name]@[version]',
+    override: process.env.EXTERNAL_CDN_TEMPLATE,
+  },
+  externalBlockList: {
+    default: [],
+    override: process.env.EXTERNAL_BLOCK_LIST
+      ? process.env.EXTERNAL_BLOCK_LIST.split(',')
```

Comment:
Maybe it's not a big deal, but I guess this lacks any validation at the moment, so could throw or lead to unexpected behaviour if someone provides unexpected values. For example something like `foo,,`.

The boolean and single-string cases are a lot simpler and I can't see anything breaking there, it's just the use of "if there is any value for X, assume I can do `split(',')` on it" that might be a touch unsafe

Reply:
good shout, I'll see if I can do some light validation. 
This is how it currently behaves in released modular, but no reason we can't improve it and make it safer

Reply:
If we have the guarantee that the value is a string (empty or not), that helps, but I wonder if ending up with empty items in the output could lead to unexpected behaviour. Maybe just check that every time in the `split()` outcome is a string with .length of >0, otherwise, potentially throw? It could be helpful to tell the user that their config might be wrong

(That said we don't want to throw when it's simply un-set or an empty string!)

Reply:
I was adding validation but I think the context of this logic makes it okay without:
Process.env always returns a string, so we're not worried about the type. 
if there's no ',' to split on, we take the whole string as a package name 
If we split on the ',' and we get empty strings, they won't match with any packages, and won't cause any damage
So the worst case scenario is that they enter the env variable incorrectly and don't get the exclusion/inclusion they wanted in their block list - but nothing will break